pkg/resource/interface.go

```diff
@@ -71,31 +71,36 @@ func (i *InterfaceFuncs) UpdateStatus(ctx context.Context, obj runtime.Object, o
 	return i.UpdateStatusFunc(ctx, obj, options)
 }
 
-func (i *InterfaceFuncs) Delete(ctx context.Context, name string,
+func (i *InterfaceFuncs[T]) Delete(ctx context.Context, name string,
 	options metav1.DeleteOptions, //nolint:gocritic // hugeParam - we're matching K8s API
 ) error {
 	return i.DeleteFunc(ctx, name, options)
 }
 
 //nolint:gocritic // hugeParam - we're matching K8s API
-func (i *InterfaceFuncs) List(ctx context.Context, options metav1.ListOptions) ([]runtime.Object, error) {
+func (i *InterfaceFuncs[T]) List(ctx context.Context, options metav1.ListOptions) ([]T, error) {
 	return i.ListFunc(ctx, options)
 }
 
 // DefaultUpdateStatus returns NotFound error indicating the status subresource isn't supported.
 //
 //nolint:gocritic // hugeParam - we're matching K8s API
-func DefaultUpdateStatus(_ context.Context, _ runtime.Object, _ metav1.UpdateOptions) (runtime.Object, error) {
-	return nil, apierrors.NewNotFound(schema.GroupResource{}, "")
+func DefaultUpdateStatus[T runtime.Object](_ context.Context, _ T, _ metav1.UpdateOptions) (T, error) {
+	return *new(T), apierrors.NewNotFound(schema.GroupResource{}, "")
```

Comment:
So can't return nil (directly) since we don't know the actual type... so allocate an instance with `*new(T)` and return the zero value, which for a pointer type is nil... cool workaround.